src/App.js

```diff
@@ -131,14 +145,20 @@ function App() {
 
   const matches = useMemo(() => {
     const nextMatches = {};
-    guesses.forEach((guess) => {
+    let newMatches = 0, newMisMatches = 0;
+    guesses.forEach((guess, guessIndex) => {
+      const isLastGuess = guessIndex === guesses.length - 1;
       indexes.forEach((index) => {
         const result = check(index, guess, word);
-        if (nextMatches[guess[index]] !== "match")
+        if (nextMatches[guess[index]] !== "match") {
           nextMatches[guess[index]] = result;
+          if (result === "match" && isLastGuess) newMatches++;
+          if (result === "mismatch" && isLastGuess) newMisMatches++;
+        };
       });
     });
-    return nextMatches;
+    console.log(newMatches, newMisMatches);
+    return {nextMatches, newMatches, newMisMatches};
```

Comment:
Ah, I see what's going on here. Calling the first property "nextMatches" threw me. That naming pattern is usually used to indicate that "nextMatches" is the next version of "matches". Instead of doing it this way, I'd recommend something like:
```
const { matches, newMatches, newMisMatches } = useMemo(() => { ... return { matches: nextMatches, newMatches, newMisMatches };}, ...);
```
That way, you can still pass `matches` around as before, and not need to change Guess to know about `matches.newMatches`. It will also keep the code a little simpler, not needing to spell out `matches.newMatches` later on.